mixed-content/generic/common.js

```diff
@@ -0,0 +1,238 @@
+  for (var attr in attrs)
+    el[attr] = attrs[attr];
+}
+
+function createElement(tagName, attrs, parent) {
+  var el = document.createElement(tagName);
+  setAttributes(el, attrs);
+  if (parent && parent.appendChild)
+    parent.appendChild(el);
+
+  return el;
+}
+
+function createHelperIframe(name) {
+  return createElement("iframe", {"name": name}, document.body);
+}
+
+function queryIframe(url) {
+  createElement("iframe", {"src": url}, document.body);
+}
+
+function queryImage(url) {
+  createElement("img", {"src": url}, document.body)
+}
+
+function queryXhr(url) {
+  var xhr = new XMLHttpRequest();
+  xhr.open('GET', url, true);
+  xhr.responseType = "json";
+  xhr.send();
+}
+
+function queryWorker(url) {
+  var worker = new Worker(url);
+  console.log("worker", url);
+  worker.postMessage('');
+}
+
+function queryFetch(url) {
+  try {
+    fetch(url);
+  } catch (ex) {
+    console.error("Failed: fetch constructor.", ex);
+  }
+}
+
+function queryNavigable(element, url) {
+  var navigable = element
+  setAttributes(navigable,
+                {"href": url,
+                 "target": createHelperIframe("helper-iframe").name});
+  navigable.click();
+}
+
+function queryAnchor(url) {
+  var a = createElement("a", {"innerHTML": "Link to resource"}, document.body);
+  queryNavigable(a, url)
+}
+
+function queryArea(url) {
+  var area = createElement("area", {}, document.body);
+  queryNavigable(area, url)
+}
+
+function queryScript(url) {
+  var script = createElement("script", {"src": url}, document.body);
+}
+
+function queryForm(url) {
+  var form = createElement("form",
+                           {"action": url,
+                            "method": "POST",
+                            "target": createHelperIframe("helper-iframe").name},
+                           document.body);
+  form.submit();
+}
+
+function queryLinkStylesheet(url) {
+  createElement("link", {"rel": "stylesheet", "href": url}, document.head);
+}
+
+function queryLinkPrefetch(url) {
+  createElement("link", {"rel": "prefetch", "href": url}, document.head);
+}
+
+function queryMedia(type, media_attrs, source_attrs) {
+  var mediaElement = createElement(type, media_attrs, document.body);
+  var sourceElement = createElement("source", source_attrs, mediaElement)
+  return mediaElement;
+}
+
+function queryVideo(url) {
+  queryMedia("video", {}, {type: "video/mp4", src: url});
+}
+
+function queryAudio(url) {
+  queryMedia("audio", {}, {type: "audio/mpeg", src: url});
+}
+
+function queryPicture(url) {
+ var picture = queryMedia("picture", {}, {"srcset": url, "type": "image/png"});
+ createElement("img", {"src": url}, picture);
+}
+
+function queryObject(url) {
+  createElement("object", {"data": url}, document.body);
+}
+
+
+function guid() {
+  function s4() {
+    return (Math.floor((Math.random() + 1.0) * 0x10000))
+           .toString(16).substring(1);
+  }
+  return s4() + s4() + '-' + s4() + '-' + s4() + '-' + s4() + '-' + s4() +
+         s4() + s4();
```

Comment:
This looks like it was taken from http://stackoverflow.com/questions/105034/create-guid-uuid-in-javascript. The second answer there seems better.